### ISSUE TITLE
Are you sure this works? for loop in a dictionary gives dictionary keys

### ISSUE THREAD

**Issue by wcafricanus** [Abhinav4382/backend-assignment#2]
https://github.com/Abhinav4382/backend-assignment/blob/05b4480254a8e298a75a12a23f648dde4622ff4c/api/views.py#L76

in line 76 when you do for loop on a dictionary `for cached_facility_data in partial_cached_data:` the `cached_facility_data` will be assigned as a dict key at every loop, which is a string in your case.
Then in the next line  `cached_facility_data["business_facility"]` you are treating `cached_facility_data` as a dictionary. Are you sure this whole thing even work? have you tested it?

**Comment by Abhinav4382** [comment]
Yes, you are right.The logic was to retrieve data for different facilities in stages and merge the results to optimize calculations. However, the code flow does not currently reach the merging logic. Caching works as expected for identical requests, but it doesn't handle partial cache retrieval and merging for varying requests.
 I’ll make sure to address this in the future.
